### ISSUE TITLE
Link prediction with Node2Vec

### ISSUE THREAD

**Issue by amarzullo24** [stellargraph/stellargraph#1897]
Dear all,
I was reading the tutorial about [link prediction using node2vec](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/node2vec-link-prediction.html#Evaluate-the-best-model-using-the-test-set). Here, Node2Vec is firstly used to [calculate node embeddings on the training set](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/node2vec-link-prediction.html#Node2Vec). This can be done using:
`embedding_train = node2vec_embedding(graph_train, "Train Graph")`

Then, in order to evaluate the performance over unseen data, we use Node2Vec to [calculate node embeddings on the test set](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/node2vec-link-prediction.html#Evaluate-the-best-model-using-the-test-set):

`embedding_test = node2vec_embedding(graph_train, "Train Graph")`

It is not clear to me why we need to learn the embeddings twice. Indeed, how can we ensure that the two learned embeddings are "comparable" (given the stochastic nature of Node2Vec). Instead, I would have expected that the model (trained on the training set) have been **applied** to the test set.

Thank you in advance for your help!

**Comment by adocherty** [comment]
Hi,

I think that you are correct - the Node2Vec embeddings will change between runs even on the same graph.  As I understand, the intention in this code is that the first part uses the embeddings learned on the train graph and the model selection set of links to select the best binary function to calculate link embeddings.  The [second part of the code ](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/node2vec-link-prediction.html#Evaluate-the-best-model-using-the-test-set) intends to use this previously selected operator to test the performance on the test set of links.  However, at this point the embeddings as you point out have changed (there is an arbitrary global rotation for all  embeddings) and the classifier should be retrained.

We will update the example to match.  Thanks for pointing this out for us!
